extension-style/src/main/java/com/mapbox/maps/extension/style/sources/generated/GeoJsonSource.kt

```diff
@@ -21,7 +25,38 @@ import com.mapbox.maps.extension.style.utils.toValue
  * @see [The online documentation](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#geojson)
  *
  */
-class GeoJsonSource(builder: Builder) : Source(builder.sourceId) {
+class GeoJsonSource(
+  builder: Builder
+) : Source(builder.sourceId) {
+
+  var geoJsonParsed = false
+  var onGeoJsonParsedListener: ((GeoJsonSource) -> Unit)? = null
+
+  private constructor(
+    builder: Builder,
+    rawGeoJson: GeoJson?,
+    onGeoJsonParsed: (GeoJsonSource) -> Unit
+  ) : this(builder) {
+    rawGeoJson?.let {
+      onGeoJsonParsedListener = onGeoJsonParsed
+      workerHandler.post {
+        val property = PropertyValue(
+          "data",
+          when(it) {
+            is Geometry -> it.toValue()
+            is FeatureCollection -> it.toValue()
+            is Feature -> it.toValue()
+            else -> ""
```

Comment:
should we log some error here.

Reply:
that situation should never happen, will add runtime exception